File: fixtures/scale/dart/large/lib/registry/suite_catalog.dart
import '../core/cipher_suite.dart';
import 'classical_registry.dart';
import 'polyalphabetic_registry.dart';
import 'stream_registry.dart';
import 'transposition_registry.dart';
import 'digraph_registry.dart';
import 'block_registry.dart';
import 'hash_registry.dart';

List<CipherSuite> allSuites(String filter) {
  final all = classicalSuites() +
      polyalphabeticSuites() +
      streamSuites() +
      transpositionSuites() +
      digraphSuites() +
      blockSuites() +
      hashSuites();
  if (filter.isEmpty) return all;
  return all.where((s) => s.name().contains(filter)).toList();
}

File: fixtures/scale/kotlin/large/src/cryptobench/registry/SuiteCatalog.kt
package cryptobench.registry

import cryptobench.core.CipherSuite

object SuiteCatalog {
    fun allSuites(filter: String): List<CipherSuite> {
        val all = buildList {
            addAll(ClassicalRegistry.suites())
            addAll(PolyalphabeticRegistry.suites())
            addAll(StreamRegistry.suites())
            addAll(TranspositionRegistry.suites())
            addAll(DigraphRegistry.suites())
            addAll(BlockRegistry.suites())
            addAll(HashRegistry.suites())
        }
        if (filter.isEmpty()) return all
        return all.filter { it.name().contains(filter) }
    }
}
